### PR TITLE
Implement GetBeaconState Endpoint

### PR DESCRIPTION
**Pull request by rauljordan**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

Follow-up on #5666, this PR implements the `GetBeaconState` `/v1/beacon/state` endpoint into Prysm gated behind the feature flag `--enable-debug-rpc`, useful for interop testing.

CC @protolambda

### CODE REVIEW

**Comment by terencechain** [comment]
A quick note on this, a prysm node by default does not keep beacon state from genesis slot to last finalized slot. If user wants to query historical state older than last finalized slot, please use flag `--enable-new-state-mgmt`

**Comment by rauljordan** [comment]
Cool I'll add a note about that

File: beacon-chain/rpc/beacon/state.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package beacon
+
+import (
+	"context"
+
+	pbp2p "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	pbrpc "github.com/prysmaticlabs/prysm/proto/beacon/rpc/v1"
+	//"google.golang.org/grpc/codes"
+	//"google.golang.org/grpc/status"
```
Comment (prestonvanloon) [inline beacon-chain/rpc/beacon/state.go:9]: ?

File: beacon-chain/rpc/beacon/state.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+)
+
+// GetBeaconState --
+func (bs *Server) GetBeaconState(
+	ctx context.Context,
+	req *pbrpc.BeaconStateRequest,
+) (*pbp2p.BeaconState, error) {
+
+	//currentSlot := bs.GenesisTimeFetcher.CurrentSlot()
```
Comment (prestonvanloon) [inline beacon-chain/rpc/beacon/state.go:18]: ?

File: beacon-chain/rpc/beacon/state.go (hunk 1 above)
Comment (rauljordan) [reply beacon-chain/rpc/beacon/state.go:9]: Unimplemented, still in progress

File: beacon-chain/rpc/beacon/state.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
+package beacon
+
+import (
+	"context"
+
+	pbp2p "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	pbrpc "github.com/prysmaticlabs/prysm/proto/beacon/rpc/v1"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	"github.com/prysmaticlabs/prysm/shared/featureconfig"
+
```
Comment (rauljordan) [inline beacon-chain/rpc/beacon/state.go:10]:
```suggestion
```

File: shared/featureconfig/flags.go [hunk 4]
```diff
@@ -150,6 +150,10 @@ var (
 		Name:  "wait-for-synced",
 		Usage: "Uses WaitForSynced for validator startup, to ensure a validator is able to communicate with the beacon node as quick as possible",
 	}
+	enableDebugRPCEndpoints = &cli.BoolFlag{
+		Name:  "enable-debug-rpc-endpoints",
```
Comment (prestonvanloon) [inline shared/featureconfig/flags.go:154]: Is a feature flag appropriate or should you make this a regular flag? 

Feature flags are temporary while you work on the feature

File: beacon-chain/node/node.go [hunk 5]
```diff
@@ -536,6 +536,7 @@ func (b *BeaconNode) registerRPCService() error {
 	slasherCert := b.cliCtx.String(flags.SlasherCertFlag.Name)
 	slasherProvider := b.cliCtx.String(flags.SlasherProviderFlag.Name)
 	mockEth1DataVotes := b.cliCtx.Bool(flags.InteropMockEth1DataVotesFlag.Name)
+  enableDebugRPCEndpoints := ctx.Bool(flags.EnableDebugRPCEndpoints.Name)
```
Comment (shayzluf) [inline beacon-chain/node/node.go:539]: somthing went wrong with the indent here

File: beacon-chain/node/node.go [hunk 6]
```diff
@@ -568,6 +569,7 @@ func (b *BeaconNode) registerRPCService() error {
 		SlasherCert:           slasherCert,
 		SlasherProvider:       slasherProvider,
 		StateGen:              b.stateGen,
+    EnableDebugRPCEndpoints: enableDebugRPCEndpoints,
```
Comment (shayzluf) [inline beacon-chain/node/node.go:572]: same here

File: beacon-chain/node/node.go [hunk 7]
```diff
@@ -608,13 +610,15 @@ func (b *BeaconNode) registerGRPCGateway() error {
 		selfAddress := fmt.Sprintf("127.0.0.1:%d", b.cliCtx.Int(flags.RPCPort.Name))
 		gatewayAddress := fmt.Sprintf("0.0.0.0:%d", gatewayPort)
 		allowedOrigins := strings.Split(b.cliCtx.String(flags.GPRCGatewayCorsDomain.Name), ",")
+    enableDebugRPCEndpoints := b.cliCtx.Bool(flags.EnableDebugRPCEndpoints.Name)
```
Comment (shayzluf) [inline beacon-chain/node/node.go:613]: same here

**Pull request merged**